### PR TITLE
Housekeeping in preparation for more features

### PR DESCRIPTION
**Pull request by esigler**
Howdy!  So, we've recently started using Trello a lot more at the place I work at, and are interested in a Lita plugin to help us out.  I'm a fan of extending and expanding existing plugins with similar functionality, instead of making yet-another-half-implementation.

So, if you're interested, here's a housekeeping PR for lita-trello - I'll be using this as a base for the additional functionality we're thinking about adding:

Future features we're considering:
- I see you have a default_list_id Redis key, but no way to set it, which would be nice (even nicer if it was per-room)
- Having a "herald" feature where the handler can receive webhooks as cards/boards are updated, and announce in a given room
- Having a way to create new boards with default invitees, since we often use Trello for ad-hoc meeting work

Overview of changes in this PR:
- Adds in RSpec test coverage (~72%, not great, but better than 0%)
- Adds in Rubocop and moves several issues over to default community style
- Refactors routes in handler to better use regex matching instead of hand-parsing later on
- Switches to localized strings, which will be helpful when adding Lita templates for rich formatting
- Adds in some documentation flare
- Provides a config for TravisCI - it's free for OSS plugins, and helpful to verify PRs such as this one pass all the tests

### CODE REVIEW

**Comment by esigler** [comment]
_bump_

**Comment by esigler** [comment]
_bump_ Any interest in this PR?

File: lib/lita/handlers/trello.rb [hunk 1]
```diff
@@ -7,90 +5,110 @@ class Trello < Handler
...
       config :board
 
-      route(/^trello\s+new\s+[^\s]+\s+[^\s]+/i, :handle_create_card, command: true, help: {
-        "trello new LIST Do the thing" => "Create a new card in LIST called 'Do the thing'"
-      })
-      def handle_create_card(r)
-        list_name = r.args[1]
-        name = r.args[2..-1].join(" ")
+      route(
+        /^trello\s+new\s+(?<list_name>\w+)\s+(?<name>.*)/i,
```
Comment (RWJMurphy) [inline lib/lita/handlers/trello.rb]: The named capture groups are much more readable :+1:

Sadly `\w+` breaks matching list names that contain spaces; `Lita::Message#args` does shell-like splitting, allowing you to e.g. `lita trello new "to do" foo` to add a card to the "to do" board. Maybe `(?<list_name>(\w+|"[^"]+"))`?

**Comment by RWJMurphy** [comment]
Other than the regression in handling list names that contain spaces, this LGTM.

**Comment by esigler** [comment]
@RWJMurphy Good catch, added!

**Comment by RWJMurphy** [comment]
:shipit:

**Pull request merged**